### ISSUE TITLE
In Online triplet selection,Validation set average loss increased

### ISSUE THREAD

**Issue by LuuYin** [adambielski/siamese-triplet#28]
Hey, thanks for your great work, but I have faced a problem.

I have noticed that with FashionMinist Dataset, in Online triplet selection, the Validation set: average loss is not decreasing during the training, on contrast, it even increased.

I have tried on my own dataset with different network(Resnet18), but face the same problem.

Do you have any idea about why this situation happens? 
Thank you very much

**Comment by thouis** [comment]
Validation loss might increase (though usually after an initial decrease).

Does validation accuracy increase?  Validation loss is not always a good proxy for accuracy.

**Comment by adambielski** [comment]
One reason for that may be overfitting, just like @thouis mentioned.
The other is that the loss is only computed on hard examples (depending on the triplet selector you use), but for some anchor-positive pair there are no negative examples that lay within the loss margin. That's why we also count the non-zero loss triplets as one of the metric. If you take these into consideration when averaging, it may turn out that the loss in fact decreases.
